src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -675,6 +675,14 @@ protected function addHookMethods(&$script)
                 $hooks[$hook.$action] = false === strpos($script, "function $hook.$action(");
             }
         }
+
+        if (null !== $this->getBehaviorContent('parentClass') ||
+            null !== ClassTools::classname($this->getBaseClass())) {
+	    $hooks['hasBaseClass'] = true;
+        } else {
+	    $hooks['hasBaseClass'] = false;
+	}
```

Comment:
Check indentation.

Is it possible to assign the expression instead of the if/else, or perhaps use an early return?